src/RadixDlt.NetworkGateway.DataAggregator/Workers/NodeWorkers/NodeTransactionFetchWorker.cs

```diff
@@ -236,10 +238,31 @@ private async Task<FetchedTransactions> FetchTransactions(long fromStateVersion,
             fromStateVersion
         );
 
-        var (apiResponse, fetchTransactionsMs) = await CodeStopwatch.TimeInMs(async () => await _transactionStreamReaderFactory().GetTransactionStream(fromStateVersion, count, token));
-        var transactions = apiResponse.Data.Transactions;
+        var (apiResponse, fetchTransactionsMs) = await CodeStopwatch.TimeInMs(async () =>
+        {
+            return await CoreApiErrorWrapper.ResultOrError<CoreClient.ApiResponse<CoreModel.StreamTransactionsResponse>, CoreModel.StreamTransactionsErrorResponse>(() =>
+            {
+                return _transactionStreamReaderFactory().GetTransactionStream(fromStateVersion, count, token);
+            });
+        });
 
-        await _observers.ForEachAsync(x => x.TransactionsFetched(NodeName, transactions, fetchTransactionsMs));
+        if (apiResponse.Failed)
+        {
+            var ex = apiResponse.FailureResponse.OriginalApiException;
+
+            if (apiResponse.FailureResponse.Details is CoreModel.RequestedStateVersionOutOfBoundsErrorDetails oob)
+            {
+                _logger.LogWarning(ex, "Requested state version past the maximal ledger state version of {MaxLedgerStateVersion}", oob.MaxLedgerStateVersion);
```

Comment:
As mentioned, this shouldn't be a warning, as it's possible to request higher than the ndoe reports it has, courtesy of `GetFirstStateVersionToFetch`. I don't think we should even log as an Info. Maybe as Debug.
We could easily get this log multiple times a second in normal use due to one or more Core APIs lagging slightly.

Reply:
Changed to Debug + return.